# Exploración

Importaciones

In [2]:
import pandas as pd


In [3]:
df = pd.read_csv('product_activity.csv')
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 8782 entries, 0 to 8781
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   user_id            8782 non-null   str    
 1   created_at         8782 non-null   str    
 2   country            8782 non-null   str    
 3   plan_type          8782 non-null   str    
 4   user_age           8028 non-null   float64
 5   post_id            8782 non-null   str    
 6   post_category      8782 non-null   str    
 7   post_created_at    8782 non-null   str    
 8   votes_received     8782 non-null   int64  
 9   user_total_posts   8782 non-null   int64  
 10  days_since_signup  8782 non-null   int64  
 11  device_type        8782 non-null   str    
dtypes: float64(1), int64(3), str(8)
memory usage: 823.4 KB


In [4]:
df.head()

,user_id,created_at,country,plan_type,user_age,post_id,post_category,post_created_at,votes_received,user_total_posts,days_since_signup,device_type
0,U01988,2025-02-18T02:07:44,PY,pro,26.0,P0008515,sport,2025-05-07T20:55:28,7,16,78,mobile
1,U00236,2025-06-22T07:49:10,BR,free,27.0,P0001023,tech,2025-09-13T20:31:06,1,9,83,web
2,U00791,2024-02-12T02:45:45,CL,free,28.0,P0003405,tech,2024-02-14T05:17:48,11,2,2,mobile
3,U01522,2024-09-22T07:06:50,US,free,16.0,P0006524,finance,2024-09-24T07:51:34,5,2,2,web
4,U01092,2025-07-18T02:27:52,PY,free,NaN,P0004665,education,2025-07-24T04:56:56,7,2,6,mobile


In [5]:
df.describe()

,user_age,votes_received,user_total_posts,days_since_signup
count,8028.000000,8782.000000,8782.000000,8782.000000
mean,27.902591,6.918356,8.324186,29.479390
std,7.547052,5.127311,6.754906,36.819928
min,16.000000,0.000000,1.000000,0.000000
25%,22.000000,3.000000,4.000000,5.000000
50%,28.000000,6.000000,6.000000,17.000000
75%,33.000000,9.000000,11.000000,40.000000
max,58.000000,74.000000,39.000000,404.000000


# Cantidad de datos nulos

In [6]:
# Contamos la cantidad de nulos, se esperan 754

total_null = df.isnull().sum()
print(total_null)

user_id                0
created_at             0
country                0
plan_type              0
user_age             754
post_id                0
post_category          0
post_created_at        0
votes_received         0
user_total_posts       0
days_since_signup      0
device_type            0
dtype: int64


# Cantidad de filas repetidas exactamente

In [7]:
# Contamos la cantidad de filas duplicadas.

duplicated_rows = df.duplicated().sum()
print(duplicated_rows)

172


In [8]:
# Controlamos que la cantidad de post_id duplicados sin que las filas sean exactas.

duplicated_post_id = df.duplicated(subset='post_id').sum()
print(duplicated_post_id)

172


# Cantidad de filas duplicadas: 
Existen 172 filas duplicadas exactas, correspondientes 1:1 con post_id repetidos → limpieza simple, sin ambigüedad

# ¿Cuántos posts ocurren antes del signup?

In [9]:
df['created_at'] = pd.to_datetime(df['created_at'], errors='coerce')
df['post_created_at'] = pd.to_datetime(df['post_created_at'], errors='coerce')
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 8782 entries, 0 to 8781
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   user_id            8782 non-null   str           
 1   created_at         8781 non-null   datetime64[us]
 2   country            8782 non-null   str           
 3   plan_type          8782 non-null   str           
 4   user_age           8028 non-null   float64       
 5   post_id            8782 non-null   str           
 6   post_category      8782 non-null   str           
 7   post_created_at    8781 non-null   datetime64[us]
 8   votes_received     8782 non-null   int64         
 9   user_total_posts   8782 non-null   int64         
 10  days_since_signup  8782 non-null   int64         
 11  device_type        8782 non-null   str           
dtypes: datetime64[us](2), float64(1), int64(3), str(6)
memory usage: 823.4 KB


In [ ]:
# controlamos cual es la fila que quedó como NaT razon por la que tenemos solo 8781 en las columnas created_at.

df[df['created_at'].isnull()]


,user_id,created_at,country,plan_type,user_age,post_id,post_category,post_created_at,votes_received,user_total_posts,days_since_signup,device_type
8772,U20001,NaT,PY,pro,29.0,P0099001,tech,2025-03-01 10:00:00,4,3,1,web


1 fecha no parseable en created_at, fila con user_id=U20001, post_id=P0099001 — posible registro corrupto/de prueba

In [11]:
# controlamos cual es la fila que quedó como NaT razon por la que tenemos solo 8781 en las columnas post_created_at.

df[df['post_created_at'].isnull()]

,user_id,created_at,country,plan_type,user_age,post_id,post_category,post_created_at,votes_received,user_total_posts,days_since_signup,device_type
8778,U20007,2025-03-15 12:00:00,MX,enterprise,38.0,P0099007,science,NaT,5,10,10,web


1 fecha no parseable en post_created_at, fila con user_id=U20007, post_id=P0099007 — posible registro corrupto/de prueba

In [15]:
df[df['post_created_at'] < df['created_at']]

,user_id,created_at,country,plan_type,user_age,post_id,post_category,post_created_at,votes_received,user_total_posts,days_since_signup,device_type
22,U01754,2025-01-17 07:05:21,US,free,39.0,P0007502,sports,2024-12-18 16:05:21,3,6,0,web
170,U00707,2024-10-07 17:32:12,MX,pro,27.0,P0003002,gaming,2024-09-27 12:32:12,13,11,0,mobile
204,U00503,2025-01-13 13:27:33,US,FREE,16.0,P0002095,health,2024-12-14 01:27:33,7,9,0,web
208,U01763,2024-12-02 12:42:25,BR,free,39.0,P0007543,music,2024-11-08 11:42:25,5,13,0,web
288,U01122,2024-04-28 09:03:25,US,pro,22.0,P0004783,education,2024-03-29 16:03:25,11,6,0,mobile
...,...,...,...,...,...,...,...,...,...,...,...,...
8281,U00349,2025-12-28 12:28:47,BR,free,26.0,P0001541,tech,2025-12-15 20:28:47,23,1,0,web
8343,U00665,2024-11-03 04:05:20,MX,free,22.0,P0002787,music,2024-10-16 18:05:20,10,1,0,web
8608,U01378,2025-06-19 18:02:45,US,pro,26.0,P0005902,travel,2025-05-20 09:02:45,8,22,0,mobile
8646,U00243,2025-02-03 13:41:27,PY,free,35.0,P0001060,finance,2025-01-31 12:41:27,2,15,0,mobile


In [16]:
posts_antes_signup = df[df['post_created_at'] < df['created_at']]
porcentaje = (len(posts_antes_signup) / len(df)) * 100
print(porcentaje)

1.1386927806877705


# Entonces ¿cuántos post_created_at son anteriores a created_at? 100 - 1.14%